src/java_bytecode/java_types.h

```diff
@@ -609,12 +609,18 @@ class java_generic_symbol_typet : public symbol_typet
   /// in the vector of generic types.
   /// \param type The type we are looking for.
   /// \return The index of the type in the vector of generic types.
-  optionalt<size_t> generic_type_index(const reference_typet &type) const
+  optionalt<size_t>
+  generic_type_index(const java_generic_parametert &type) const
```

Comment:
Please update the documentation to reflect that only a generic parameter can be passed here. Also, wherever this function is called, make sure that it's only called on a parameter - for example by adding preconditions if it's not obvious.

Reply:
The type already ensures it is only called on a java_generic_parametert so I don't think a precondition is necessary

Reply:
I meant a precondition before a call to this function, elsewhere in the code.

Reply:
If the function is called then the argument is a java_generic_parametert so there shouldn't be anything to check?

Reply:
Yes, that's exactly what I meant - if the object that this function is called with is already typed as `java_generic_parametert` then there is nothing to check.